distributed/joblib.py

```diff
@@ -40,6 +41,63 @@
                        " or `sklearn` > '0.17.1'. Please install or upgrade")
 
 
+def is_weakrefable(obj):
+    try:
+        weakref.ref(obj)
+        return True
+    except TypeError:
+        return False
+
+
+class _WeakKeyDictionary:
```

Comment:
Any reason not to subclass from [MutableMapping](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping), which would turn this into a full `dict` interface?

Reply:
I would need to implement `__delitem__` (easy) and `__iter__` not as simple but we have no use for those. So I don't think it's worth it.